src/ConnectionInterface.php

```diff
@@ -208,17 +208,20 @@ public function getServerOptions();
     public function getState();
 
     /**
-     * Close the connection.
+     * Quits (soft-close) the connection.
      *
-     * @param callable|null $callback A callback which should be run after
-     *                                connection successfully closed.
-     *
-     * $callback signature:
+     * This method returns a promise that will resolve with a boolean `true` on
+     * success or will reject with an `Exception` on error. The MySQL protocol
+     * is inherently sequential, so that all commands will be performed in order
+     * and outstanding commands will be put into a queue to be executed once the
+     * previous commands are completed.
      *
-     *  function (ConnectionInterface $conn): void
+     * ```php
+     * $connection->query('CREATE TABLE test ...');
+     * $connection->quit();
+     * ```
      *
-     * @return void
-     * @throws Exception if the connection is not initialized or already closed/closing
+     * @return PromiseInterface Returns a Promise<true,Exception>
```

Comment:
I'd personally keep the "return" value as `void` or is there a reason this now resolves with `true`?

Reply:
Good point! Updated the resolve with a void value instead. I'll make sure to file a similar PR which makes sure `ping()` also resolves with a void value instead of boolean true (refs #63). :shipit:

Reply:
> I'll make sure to file a similar PR which makes sure ping() also resolves with a void value instead of boolean true (refs #63). :shipit:

See #66 :shipit: